src/SmoFacade/Server.cs

```diff
@@ -225,6 +225,28 @@ public Database Database(string dbName)
         policy.Execute(() => restore.SqlRestore(_server));
         restore.Devices.Remove(backupDeviceItem);
       }
+
+      if (fileRelocation != null)
+        LogicalFileRename(databaseName, fileRelocation);
+    }
+
+    private void LogicalFileRename(string databaseName, Func<string, string> fileRelocation)
+    {
+      var db = Database(databaseName);
+      if (db.Restoring)
+        db.RestoreWithRecovery();

```

Comment:
We don't want to finalize the `Restore` operation here.  

Notice how we finalize based on this property, which could be... false!
https://github.com/factset/AgDatabaseMove/blob/master/src/AgDatabaseMove.cs#L83

Perhaps this is here because you can only rename the logicalfilename if the DB is restored?  
To note, this rename logic doesn't necessarily need to be deeply nested here in the restore code.  
Now that this is a standalone function on `Server` - make it public/internal... and call it from other classes where applicable (ie. [here](https://github.com/factset/AgDatabaseMove/blob/master/src/AgDatabaseMove.cs#L85) )

Reply:
I can't drop our FDS code in this public repo... take a look at example usage in  FactSet.Sqlis.Core/Move.cs#L44-L68

Reply:
Yes, I threw it in there because you can only access the file names once the DB is restored. 

I'll move forward with making the function public. Given that information, I think I'll throw an exception rather than finalize the restore on L237 to ensure finalizing is done intentionally.